tests/test_utils.py

```diff
@@ -825,3 +826,28 @@ def test_non_operations_in_list(self, arg):
             ValueError, match="The given operation_list does not only contain Operations"
         ):
             pu.inv(arg)
+
+
+class TestHashIterable:
+    """Tests for the _hash_iterable function."""
+
+    iterables = [
+        [1, 1.4, -1],
```

Comment:
Could be worth adding unsupported cases:

> The iterable must be flat and can contain only numbers and NumPy arrays.

When this condition is not satisfied.